File: src/go/pe_0006.go
package main


// FUNCTIONS


func sum(arr []int) int {
    gross := 0
    for _, n := range arr {
        gross += n
    }
    return gross
}

func sum_of_squares(arr []int) int {
    gross := 0
    for _, n := range arr {
        gross += n * n
    }
    return gross
}

func square_of_sum(arr []int) int {
    gross := sum(arr)
    return gross * gross
}


// SCRIPT


func main() {
    arr := []int{}
    for n := range [100]int{} {
        arr = append(arr, n + 1)
    }
    println(square_of_sum(arr) - sum_of_squares(arr))
}

File: src/c/pe_0006.c
#include <stdio.h>


// FUNCTIONS


int sum_of_squares(int arr[], int n) {
    int gross = 0;
    for (int i = 0; i < n; ++i) {
        gross += arr[i] * arr[i];
    }
    return gross;
}


int square_of_sum(int arr[], int n) {
    int gross = 0;
    for (int i = 0; i < n; ++i) {
        gross += arr[i];
    }
    return gross * gross;
}


// SCRIPT


int main() {
    int n = 100;
    int arr[100];

    for (int i = 0; i < n; ++i) {
        arr[i] = i + 1;
    }

    printf("%d\n", square_of_sum(arr, n) - sum_of_squares(arr, n));
}
